mountpoint-s3/tests/fs.rs

```diff
@@ -445,3 +445,38 @@ async fn test_stat_block_size() {
     assert_eq!(lookup.attr.blocks, 9);
     assert_eq!(lookup.attr.blksize, 4096);
 }
+
+#[test_case("foo"; "remove file")]
+#[test_case("bar/foo"; "remove directory")]
+#[tokio::test]
+async fn test_lookup_removes_old_children(key: &str) {
```

Comment:
Can we also test that `readdir` has the same effect?

Reply:
We have not changed `readdir` behavior: it only updates keys/prefixes as they are returned, but it does not remove any "missing" children. In order to do it, I think it would need to keep track of all the "names" returned across all the `list_objects` calls in order to work out what to remove at the end.

I'm not sure though: do we need `readdir` to do it? Or can we just rely on lookup?

Reply:
Oh, right, we'll never test the negative case through `readdir`.